packages/flutter/lib/src/cupertino/refresh.dart

```diff
@@ -359,43 +357,53 @@ class CupertinoSliverRefreshControl extends StatefulWidget {
     return state.refreshState;
   }
 
-  /// Builds a simple refresh indicator that fades in a bottom aligned down
-  /// arrow before the refresh is triggered, a [CupertinoActivityIndicator]
-  /// during the refresh and fades the [CupertinoActivityIndicator] away when
-  /// the refresh is done.
-  static Widget buildSimpleRefreshIndicator(
+  /// Builds a refresh indicator that reflects the standard iOS pull-to-refresh
+  /// behavior. Specifically, this entails presenting an activity indicator that
+  /// changes depending on the current refreshState. As the user initially drags
+  /// down, the indicator will gradually reveal individual ticks until the refresh
+  /// becomes armed. At this point, the animated activity indicator will begin rotating.
+  /// Once the refresh has completed, the activity indicator shrinks away as the
+  /// space allocation animates back to closed.
+  static Widget buildRefreshIndicator(
     BuildContext context,
     RefreshIndicatorMode refreshState,
     double pulledExtent,
     double refreshTriggerPullDistance,
     double refreshIndicatorExtent,
   ) {
-    const Curve opacityCurve = Interval(0.4, 0.8, curve: Curves.easeInOut);
+    final double percentageComplete = min(pulledExtent / refreshTriggerPullDistance, 1.0);
     return Align(
-      alignment: Alignment.bottomCenter,
+      alignment: Alignment.topCenter,
       child: Padding(
-        padding: const EdgeInsets.only(bottom: 16.0),
-        child: refreshState == RefreshIndicatorMode.drag
-            ? Opacity(
-                opacity: opacityCurve.transform(
-                  min(pulledExtent / refreshTriggerPullDistance, 1.0)
-                ),
-                child: Icon(
-                  CupertinoIcons.down_arrow,
-                  color: CupertinoDynamicColor.resolve(CupertinoColors.inactiveGray, context),
-                  size: 36.0,
-                ),
-              )
-            : Opacity(
-                opacity: opacityCurve.transform(
-                  min(pulledExtent / refreshIndicatorExtent, 1.0)
-                ),
-                child: const CupertinoActivityIndicator(radius: 14.0),
-              ),
+        padding: const EdgeInsets.only(top: 16.0),
+        child: _buildIndicatorForRefreshState(refreshState, 14.0, percentageComplete),
       ),
     );
   }
 
+  static Widget _buildIndicatorForRefreshState(RefreshIndicatorMode refreshState, double radius, double percentageComplete) {
+    switch (refreshState) {
+      case RefreshIndicatorMode.drag:
+        // While we're dragging, we draw individual segments of the spinner while simultaneously
+        // easing the opacity in.
+        const Curve opacityCurve = Interval(0.0, 0.2, curve: Curves.easeInOut);
```

Comment:
Describe also in your code comments how you derived these values. (i.e. when the next maintainer comes to change this, how does that person have more or less confidence that that person applied more rigor in deriving these values than the existing values?)

Reply:
mmm... mostly it was me just stepping through a recorded video frame by frame watching for when the fade-in finished. There wasn't much science to it...  is that still reasonable to document?

Reply:
I don't want to answer for @xster, but I would say it's definitely valuable to document that. "These values were obtained by stepping through a screen recording of \<native widget\> on \<device\> running \<iOS version\>."

I have lots of comments floating around stating that I just loosely eyeballed some value. It's valuable because if someone comes along with a value they got with a more precise method, they can be confident that their value is better than mine and they should update it.

Reply:
Yes, I believe so. Note that 0.2 isn't controlling the final opacity... it is shaping the curve so that that the opacity is 1.0 by the time the drag is 20% complete. See example below.

![example_ios_spinner](https://user-images.githubusercontent.com/1574429/84445344-20424d80-ac87-11ea-808c-84b00b9c9fd7.gif)

Reply:
The accurate end value seems to be 0.35. The value is the duration of the `CABasicAnimation` that changes the opacity.

Reply:
OK, no problems. I'll update to use that value... how did you find this, so that I can also update the comments about how we derived that value? Ta

Reply:
The value was extracted using Xcode, from inspecting a native app running on iOS 13.5.